src/voltorb_flip/voltorb_flip_input.c

```diff
@@ -4,237 +4,321 @@
+    { .rect = {  80, 104, 224, 248 } },
+    { .rect = { 104, 128, 200, 224 } },
+    { .rect = { 104, 128, 224, 248 } },
+    { .rect = { 128, 152, 224, 248 } },
+    { .rect = {   8,  72, 192, 256 } },
+    { .rect = { 160, 192, 192, 256 } },
+    { .rect = TOUCHSCREEN_RECTLIST_END },
+};
+
+const TouchscreenHitbox sTouchscreenHitboxes[] = {
+    { .rect = {   8,  32,   8,  32 } },
+    { .rect = {   8,  32,  40,  64 } },
+    { .rect = {   8,  32,  72,  96 } },
+    { .rect = {   8,  32, 104, 128 } },
+    { .rect = {   8,  32, 136, 160 } },
+    { .rect = {  40,  64,   8,  32 } },
+    { .rect = {  40,  64,  40,  64 } },
+    { .rect = {  40,  64,  72,  96 } },
+    { .rect = {  40,  64, 104, 128 } },
+    { .rect = {  40,  64, 136, 160 } },
+    { .rect = {  72,  96,   8,  32 } },
+    { .rect = {  72,  96,  40,  64 } },
+    { .rect = {  72,  96,  72,  96 } },
+    { .rect = {  72,  96, 104, 128 } },
+    { .rect = {  72,  96, 136, 160 } },
+    { .rect = { 104, 128,   8,  32 } },
+    { .rect = { 104, 128,  40,  64 } },
+    { .rect = { 104, 128,  72,  96 } },
+    { .rect = { 104, 128, 104, 128 } },
+    { .rect = { 104, 128, 136, 160 } },
+    { .rect = { 136, 160,   8,  32 } },
+    { .rect = { 136, 160,  40,  64 } },
+    { .rect = { 136, 160,  72,  96 } },
+    { .rect = { 136, 160, 104, 128 } },
+    { .rect = { 136, 160, 136, 160 } },
+    { .rect = {   8,  72, 192,   0 } },
+    { .rect = { 160, 192, 192,   0 } },
+    { .rect = TOUCHSCREEN_RECTLIST_END },
+};
+
+const DpadMenuBox sDpadButtonSpecs[] = {
+    {   8,   8, 24, 24, 20,  5, 25, 1  },
+    {  40,   8, 24, 24, 21,  6,  0, 2  },
+    {  72,   8, 24, 24, 22,  7,  1, 3  },
+    { 104,   8, 24, 24, 23,  8,  2, 4  },
+    { 136,   8, 24, 24, 24,  9,  3, 25 },
+    {   8,  40, 24, 24,  0, 10, 25, 6  },
+    {  40,  40, 24, 24,  1, 11,  5, 7  },
+    {  72,  40, 24, 24,  2, 12,  6, 8  },
+    { 104,  40, 24, 24,  3, 13,  7, 9  },
+    { 136,  40, 24, 24,  4, 14,  8, 25 },
+    {   8,  72, 24, 24,  5, 15, 25, 11 },
+    {  40,  72, 24, 24,  6, 16, 10, 12 },
+    {  72,  72, 24, 24,  7, 17, 11, 13 },
+    { 104,  72, 24, 24,  8, 18, 12, 14 },
+    { 136,  72, 24, 24,  9, 19, 13, 25 },
+    {   8, 104, 24, 24, 10, 20, 26, 16 },
+    {  40, 104, 24, 24, 11, 21, 15, 17 },
+    {  72, 104, 24, 24, 12, 22, 16, 18 },
+    { 104, 104, 24, 24, 13, 23, 17, 19 },
+    { 136, 104, 24, 24, 14, 24, 18, 26 },
+    {   8, 136, 24, 24, 15,  0, 26, 21 },
+    {  40, 136, 24, 24, 16,  1, 20, 22 },
+    {  72, 136, 24, 24, 17,  2, 21, 23 },
+    { 104, 136, 24, 24, 18,  3, 22, 24 },
+    { 136, 136, 24, 24, 19,  4, 23, 26 },
+    { 224,  40,  0,  0, 25, 26, 27, 28 },
+    { 224, 176,  0,  0, 25, 26, 27, 28 },
+};
+
+u8 sMemoCursorPos[5][2] = {
+    { 200,  80 },
+    { 224,  80 },
+    { 200, 104 },
+    { 224, 104 },
+    { 224, 128 },
+};
+// clang-format on
+
+VoltorbFlipInputHandler *VoltorbFlipInputHandler_Create(enum HeapID heapID, struct ManagedSprite *a1, struct ManagedSprite *a2) {
```

Comment:
Documentation: name a1, a2

Reply:
Couldn't come up with good names for these on my own.